sdk/core/azure-core-metrics-opentelemetry/src/main/java/com/azure/core/metrics/opentelemetry/OpenTelemetryUtils.java

```diff
@@ -30,8 +30,16 @@ static void addAttribute(io.opentelemetry.api.common.AttributesBuilder attribute
             attributesBuilder.put(AttributeKey.stringKey(key), (String) value);
         } else if (value instanceof Double) {
             attributesBuilder.put(AttributeKey.doubleKey(key), (Double) value);
+        } else if (value instanceof Float) {
+            attributesBuilder.put(AttributeKey.doubleKey(key), ((Float) value).doubleValue());
```

Comment:
You're casting to Float, but then putting a double into the builder?

Reply:
OTel does not support float attributes, so I have to cast it to double  this way:(